### PR TITLE
Edit Vert/Edge/Face shortcuts now auto-enter PB context

### PR DESCRIPTION
**Pull request by modrimkus-unity**
### Purpose of this PR

"Edit Vertices/Edges/Faces" shortcuts now automatically enter the ProBuilder context if that's possible. This is part of the 3 step fix where we remove PB's context switch shortcut, add this change and update trunk so that context cycle shortcut include game object context.

### Links

**Jira:** https://jira.unity3d.com/browse/PBLD-169

### Comments to Reviewers

[List known issues, planned work, provide any extra context for your code.]

### CODE REVIEW

File: Editor/EditorCore/ProBuilderEditorShortcuts.cs [hunk 1]
```diff
@@ -43,5 +52,19 @@ static void Toggle_SelectMode()
 			if(ProBuilderEditor.instance != null)
 				ProBuilderEditor.instance.ToggleSelectionMode();
 		}
+
+		static bool CheckAndEnterPBContextIfNeeded()
+		{
+			if (ToolManager.activeContextType == typeof(GameObjectToolContext))
```
Comment (lopezt-unity) [inline Editor/EditorCore/ProBuilderEditorShortcuts.cs:58]: I wonder if that should be triggerable from any context as long as a PB GO is selected (to check with Gabe)?
Like : "I have GOs with Splines and PB components, I'm in the spline context and want to directly enter Vertex edit mode" ?
```suggestion
			if (ToolManager.activeContextType != typeof(PositionToolContext))
```

Comment (modrimkus-unity) [reply Editor/EditorCore/ProBuilderEditorShortcuts.cs:58]: Checking with @gabrielw-us

**Review by claudiaantoun — APPROVED** [review]

File: Editor/EditorCore/ProBuilderEditorShortcuts.cs (hunk 1 above)
Comment (modrimkus-unity) [reply Editor/EditorCore/ProBuilderEditorShortcuts.cs]: @lopezt-unity Applied the suggestion after confirming with Gabe.

**Review by lopezt-unity — APPROVED** [review]
LGTM

**Pull request merged**